### PR TITLE
Adding support for where argument

### PR DESCRIPTION
**Pull request by soma2000-lang**
In continuation from #14665
Functions added for

logaddex
isnan
isinf
log
less
less_equal
log10
log1p
log2

### CODE REVIEW

**Comment by soma2000-lang** [comment]
@VedPatwardhan I am generating the where argument in the test ,by  '''where=draw(st.lists(st.booleans(), min_size=1, max_size=100))''' because where is basically an array of boolean mask,but still the tests are fmiailing.I am facing difficulty in understanding what am I missing here.

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hey @soma2000-lang, I think the tests might be failing because the input array and the boolean mask (i.e. `where`) should have broadcastable shapes, which isn't guaranteed in the data generation of the `where` argument. Could you please update it so that the input arrays and `where` are broadcastable? Thanks @soma2000-lang 🙂

**Comment by soma2000-lang** [comment]
Hi @vedpatwardhan  One update reagrding this issue ,I have updated my the tests for addition of the where argument and I dont think the failures are due to this addition ,as seen locally
image

File: ivy_tests/test_ivy/test_functional/test_core/test_elementwise.py [hunk 1]
```diff
@@ -971,19 +971,30 @@ def test_isfinite(
...
 
 
+@st.composite
+def _ishelp(draw):
+    dtype, x, shape = draw(
+        helpers.dtype_and_values(
+            available_dtypes=helpers.get_dtypes("numeric"), ret_shape=True
+        )
+    )
+    dtype2, where = draw(
```
Comment (vedpatwardhan) [inline ivy_tests/test_ivy/test_functional/test_core/test_elementwise.py:981]: I think the test should be made more exhaustive by generating `where` argument of all broadcastable shapes from `shape`. We can use helpers like [this](https://github.com/unifyai/ivy/blob/5d237bccc2ca36927b3f170c81aae331a7406d47/ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py#L1698) or [this](https://github.com/unifyai/ivy/blob/5d237bccc2ca36927b3f170c81aae331a7406d47/ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py#L1670) to generate such shapes

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]
> Hi @vedpatwardhan One update regarding this issue,I have updated the tests for the addition of the where argument and I don't think the failures are due to this addition, as seen locally, you can check the PR once for the updated where implementation.I have tried to fix some tests for the torch backend but still there are failures remaining. 

Hey @soma2000-lang, just requested a few changes, yeah the `KeyError` issue isn't related to your PR and is something being fixed at the moment. Thanks 🙂

File: ivy_tests/test_ivy/test_functional/test_core/test_elementwise.py [hunk 2]
```diff
@@ -971,19 +971,30 @@ def test_isfinite(
     )
 
 
+@st.composite
+def _ishelp(draw):
```
[the remark this answers is not in the document]
Comment (soma2000-lang) [reply ivy_tests/test_ivy/test_functional/test_core/test_elementwise.py:975]: Sure!

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Pull request closed without merging**